nemo/collections/llm/modelopt/model_utils.py

```diff
@@ -48,7 +48,7 @@ def _set_gpt_modelopt_spec(model_cfg: llm.GPTConfig) -> llm.GPTConfig:
 
 def set_modelopt_spec_if_exists_in_ckpt(model: L.LightningModule, path: str) -> None:
     """Set model.config.transformer_layer_spec to modelopt spec if modelopt_state exists in the checkpoint."""
-    path = str(path).lstrip("nemo://")  # Remove nemo:// prefix added by finetune_recipe
+    path = str(path).removeprefix("nemo://")  # Remove nemo:// prefix added by finetune_recipe
```

Comment:
during SFT `lstrip` produces a bug during checkpoint loading since it removes the trailing `/` from the file-path. then `ckpt_to_weights_subdir` does not recognize the `path` as a directory